src/modules/tokens/hooks/useEnoughBalance.ts

```diff
@@ -0,0 +1,66 @@
+import { Currency, CurrencyAmount, Token } from '@uniswap/sdk-core'
+
+import useNativeCurrency from 'lib/hooks/useNativeCurrency'
+import { getAddress } from 'utils/getAddress'
+import { isEnoughAmount } from 'utils/isEnoughAmount'
+
+import { useBalancesAndAllowances } from './useBalancesAndAllowances'
+import { useCurrencyBalances } from './useCurrencyBalance'
+
+import { TokenAmounts } from '../types'
+
+export interface UseEnoughBalanceParams {
+  account?: string
+  amount?: CurrencyAmount<Currency>
+  checkAllowanceAddress?: string
+}
+
+export function useEnoughBalanceAndAllowance(params: UseEnoughBalanceParams): boolean {
+  const { account, amount, checkAllowanceAddress } = params
+  const isNativeCurrency = amount?.currency.isNative
+  const token = amount?.currency.wrapped
+
+  const { balances, allowances } = useBalancesAndAllowances({
+    account,
+    spender: checkAllowanceAddress,
+    tokens: !isNativeCurrency && account && token ? [token as Token] : [],
```

Comment:
Doesn't `currency.wrapped` return a `Token` instance already?
Meaning, is the cast needed?

Reply:
But i don't want the wrapped ETH, i want tokens to be EMPTY list if it's ETH, because in this case we check the Ether balances, not ERC20 balances.

This hook will handle this specific case.

Reply:
> But i don't want the wrapped ETH, i want tokens to be EMPTY list if it's ETH,
You already have `!isNativeCurrency` in the conditional, so you will know

Reply:
Anyway, I see you are already using the `wrapped` instance, so you don't need the cast anymore, no?